GEODE-2014: Upgrade Swagger libraries

kjduling opened the pull request.

metatype commented on geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestSecurityIntegrationTest.java:
```diff
@@ -94,7 +94,7 @@ public static void before() throws Exception {
   public void testFunctions() throws Exception {
     String json = "{\"@type\":\"double\",\"@value\":210}";
-
+    Thread.sleep(500000);
```
This seems like a really long pause for an integration test.  Is there another way to do this?

kirklund replied:
Yeah, I think we should be very adamant about not adding sleeps. Please change this to an Awaitility call that's awaiting whatever condition this sleep is trying to let happen.

kjduling replied:
Whoops!  That shouldn't have gone in.  I needed that for hand-testing swagger.

kirklund commented on gradle/dependency-versions.properties:
```diff
@@ -100,8 +100,8 @@ spring-tx.version = 4.3.2.RELEASE
 spymemcached.version = 2.9.0
-swagger.version = 1.3.2
-swagger-springmvc.version = 0.8.2
+swagger.version=1.3.13
```
However, 1.3.13 is an old Swagger release. I don't think we should upgrade to an old version. If we're going to upgrade then let's upgrade to the latest version (1.5.10?).

kjduling commented:
Added a test to see if Swagger is up and the splash page has the right Geode headers.
Upgraded to the Swagger2 specification
Removed the accidental checkin of the sleep.

jinmeiliao commented on geode-assembly/src/test/java/org/apache/geode/rest/internal/web/SwaggerVerificationTest.java:
```diff
@@ -0,0 +1,57 @@
+    assertThat(getCode(response), is(200));
+
+    // Check the JSON
+    response = doRequest(new HttpGet("/geode/v2/api-docs"), "", "");
```
Just curious, why this is geode/v2, but all the api is still geode/v1

kjduling replied:
"geode/v2/api-docs" is the endpoint for Swagger2 documentation.  Swagger was "geode/api-docs".  Geode's endpoints are all "geode/v1/*" and I've verified that if/when the REST API revision is bumped to "v2", there will be no conflict unless we create a new endpoint called "v2/api-docs".

This is supposed to be able to be overridden with the "springfox.documentation.swagger.v2.path" env property, but it doesn't seem to be honored.  Or SwaggerConfig is being loaded too late in the lifecycle, which I doubt.  At any rate, I am adding the structure to use this, but leaving the values set to the default ones.

jinmeiliao commented on geode-web-api/src/main/java/org/apache/geode/rest/internal/web/controllers/FunctionAccessController.java:
```diff
@@ -93,8 +93,7 @@ protected String getRestApiVersion() {
   @ApiResponses({
-      @ApiResponse(code = 200, message = "OK."),
-      @ApiResponse( code = 401, message = "Invalid Username or Password." ),
+      @ApiResponse(code = 200, message = "OK."), @ApiResponse(code = 401, message = "Invalid Username or Password."),
```
use the old fomat?
